src/platform/unix/mod.rs

```diff
@@ -649,10 +649,69 @@ fn make_socket_lingering(sockfd: c_int) -> Result<(),UnixError> {
     Ok(())
 }
 
+struct BackingStore {
+    fd: c_int
+}
+
+impl BackingStore {
+    pub unsafe fn create(length: usize) -> BackingStore {
```

Comment:
Good point. I renamed the function to `new` and moved the code inside of an `unsafe` block. I probably could have moved only parts of the code inside the `unsafe` block, but I thought this looked more readable. I'm okay with changing that though.